### ISSUE TITLE
Validation of projectname is done after selecting routing/style

### ISSUE THREAD

**Issue by mhombach** [angular/angular-cli#17595]
# 🐞 Bug report

### Command (mark with an `x`)

- [X] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

I think this bug was existing for along time.

### Description
If i create a new project by `ng new xyz` and the projectname is not valid (for example `test_project` because of the underscore), the CLI is telling me about the invalid name only AFTER i had to select angular-routing (y/N) and what style-type i want to use (css/scss/...).
This should be changed in a way where the proejctname is validated FIRST, before i have to choose other things.
If the selections i make could have an effect on the validation, that would be a different story, but if the projectname is invalid, there is nothing in those 2 selections that could make it working.

## 🔬 Minimal Reproduction
```
ng new test_project
<select Y/N for routing>
<select style type>
```

## 🔥 Exception or Error
<pre><code>
Project name "test_project" is not valid. New project names must start with a letter, and must contain only alphanumeric characters or dashes. When adding a dash the segment after the dash must also start with a letter.
</code></pre>

## 🌍 Your Environment
<pre><code>
Angular CLI: 9.1.1
Node: 13.3.0
OS: win32 x64

Angular:
...
Ivy Workspace:

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.901.1
@angular-devkit/core         9.1.1
@angular-devkit/schematics   9.1.1
@schematics/angular          9.1.1
@schematics/update           0.901.1
rxjs                         6.5.4
</code></pre>

**Anything else relevant?**

**Comment by alan-agius4** [comment]
Duplicate of https://github.com/angular/angular-cli/issues/14994

**Comment by mhombach** [comment]
Searched for similar issues before, sorry, it seems i used different words for searching :/

**Comment by alan-agius4** [comment]
No worries @mhombach 😁
